### ISSUE TITLE
NerDLApproach no longer accepts DBFS path for graph folder

### ISSUE THREAD

**Issue by katelynrwalker** [JohnSnowLabs/spark-nlp#12611]
## Description

This has been previously addressed in issue [#739](https://github.com/JohnSnowLabs/spark-nlp/issues/739), but it seems that some more recent release may have caused regression.

In Spring 2021, I trained an NerDL model using spark-nlp 3.3.0 and a Databricks runtime that I didn't record. At that time, I was able to access custom graphs using the following:

`nerTagger = NerDLApproach().setGraphFolder(file:///dbfs/katelyn/graphs/)`

When I try to retrain the same model today on spark-nlp 4.0.2, with the custom graphs in the same location, I get the error:  "java.io.FileNotFoundException: folder: file:///dbfs/katelyn/graphs not found"

## Expected Behavior

NerDLApproach().setGraphFolder should be able to access custom graphs in the DBFS file system

## Current Behavior

I have tried a number of possible variations on the file path, and continue to get various errors:

- file:///katelyn/graphs/ (folder not found)
- /katelyn/graphs/ (folder not found)
- /dbfs/katelyn/graphs (folder not found)
- dbfs:/katelyn/graphs/ (gives error "IllegalArgumentException: Wrong FS: dbfs:/katelyn/graphs/blstm_125_768_128_96.pb, expected: file:///")
- katelyn/graphs (gives error that path must be absolute)
- dbfs:/mnt/tmp/graphs/ (for this one, moved the graph folders to the mnt/tmp location, as suggested in a comment on issue #739; gives error "IllegalArgumentException: Wrong FS: dbfs:/mnt/tmp/graphs/blstm_125_768_128_96.pb, expected: file:///")

Additionally, it would be great if this could fail more quickly - right now, the error does not occur until after the pre-saved embeddings data is fully loaded, which takes ~1.5 hours for my full dataset. Even running on a test set of 1 sample, it takes several minutes to fail.

## Possible Solution
Spark NLP 4.0.x is based on TensorFlow 2.7.x, but these graphs were created using TensorFlow 1.15.0. It's entirely possible this is the real problem, and these folder-not-found errors are a red herring. 

However it doesn't appear that the script for creating custom graphs has been updated for TF 2.7.x: https://nlp.johnsnowlabs.com/docs/en/graph. When I try to create a new graph on DBR 10.4, I get errors such as "Exception: Python 3.7 or above not supported by TensorFlow" and "module 'tensorflow' has no attribute 'disable_v2_behavior'" (based on whether I run the bash script or copy and paste the function into my notebook).

## Steps to Reproduce

1. Using notebook attached to either interactive cluster or Databricks job running 10.4ML, with Spark-NLP 4.0.2:
2. (This part works fine):
```
from sparknlp.training import CoNLL
training_data = CoNLL().readDataset(spark, 's3a://dataPath')
embeddings = BertEmbeddings.pretrained('bert_base_cased', 'en') \
.setInputCols("sentence","token")\
.setOutputCol("embeddings")\
embeddedDF = embeddings.transform(training_data)
embeddedDF.write.mode("overwrite").parquet("savePath")
```

3. This is where the error occurs:
```
embeddedDF2 = sqlContext.read.parquet("savePath")
nerTagger = NerDLApproach()\
.setInputCols(["sentence", "token", "embeddings"])\
.setLabelColumn("label")\
.setOutputCol("ner")\
.setMaxEpochs(10)\
.setRandomSeed(0)\
.setBatchSize(32)\
.setVerbose(1)\
.setValidationSplit(0.1)\
.setEvaluationLogExtended(True)\
.setEnableOutputLogs(True)\
.setIncludeConfidence(True)\
.setEnableMemoryOptimizer(True)\
.setGraphFolder("dbfs:/mnt/tmp/citadel_graph/")
ner_model = nerTagger.fit(embeddedDF2)
```

## Context
Unable to train an NerDL model with custom graph

## Your Environment

* Spark NLP version `sparknlp.version()`: 4.0.2
* Databricks runtime: 10.4 ML LTS
* Apache NLP version `spark.version`: 3.1.2

**Comment by maziyarpanahi** [comment]
thanks @katelynrwalker for reporting this
>Spark NLP 4.0.x is based on TensorFlow 2.7.x, but these graphs were created using TensorFlow 1.15.0. It's entirely possible this is the real problem, and these folder-not-found errors are a red herring.

The error is regarding the access/list of that specific directory on the DBFS file system, it is not really about the TF version and all the graphs are compatible from 1.15 all the way to 2.7. (also, the error for the TF graph will be from TensorFlow, this seems to be a problem in accessing DBFS which at some point might have been broken)

@danilojsl could you please look into this to see if we might have broken something by adding something new?

**Comment by danilojsl** [comment]
@maziyarpanahi @katelynrwalker it seems that is not working when defining a prefix different than `dbfs:/` I tried with `/dbfs/` and got the error `java.io.FileNotFoundException: file or folder: /dbfs/danilo_tmp/graphs/ner-dl/ not found`

However, in 4.0.1 defining the graph path with `dbfs:/` prefix works. It is important to stand out that since spark-nlp 4.1.0, it also works with prefix `/dbfs/`.

I tested it with spark-nlp 4.0.2 and 4.1.0 in Scala and Python with the following Databricks Runtime Versions:
10.4LTS Apache Spark 3.2.1
10.4 Apache Spark 3.2.1
9.1 LTS Spark 3.1.2

@katelynrwalker could you please retry using `dbfs:/` or test in spark-nlp 4.1.0 +

**Comment by danilojsl** [comment]
@maziyarpanahi since there is no issue in 4.1 and 4.2 using `dbfs:/`, I will close the issue.
